src/spikeinterface/core/template.py

```diff
@@ -0,0 +1,156 @@
+        if self.channel_ids is None:
+            self.channel_ids = np.arange(self.num_channels)
+        if self.unit_ids is None:
+            self.unit_ids = np.arange(self.num_units)
+        if self.sparsity_mask is not None:
+            self.sparsity = ChannelSparsity(
+                mask=self.sparsity_mask,
+                unit_ids=self.unit_ids,
+                channel_ids=self.channel_ids,
+            )
+
+    def to_dict(self):
+        return {
+            "templates_array": self.templates_array.tolist(),
+            "sparsity_mask": None if self.sparsity_mask is None else self.sparsity_mask.tolist(),
+            "channel_ids": self.channel_ids.tolist(),
+            "unit_ids": self.unit_ids.tolist(),
+            "sampling_frequency": self.sampling_frequency,
+            "nbefore": self.nbefore,
+        }
+
+    @classmethod
+    def from_dict(cls, data):
+        return cls(
+            templates_array=np.array(data["templates_array"]),
+            sparsity_mask=None if data["sparsity_mask"] is None else np.array(data["sparsity_mask"]),
+            channel_ids=np.array(data["channel_ids"]),
+            unit_ids=np.array(data["unit_ids"]),
+            sampling_frequency=data["sampling_frequency"],
+            nbefore=data["nbefore"],
+        )
+
+    def get_dense_templates(self) -> np.ndarray:
+        # Assumes and object without a sparsity mask already has dense templates
+        if self.sparsity is None:
+            return self.templates_array
+
+        dense_waveforms = np.zeros(shape=(self.num_units, self.num_samples, self.num_channels))
+        for unit_index, unit_id in enumerate(self.unit_ids):
+            num_active_channels = self.sparsity.mask[unit_index].sum()
+            waveforms = self.templates_array[unit_index, :, :num_active_channels]
+            dense_waveforms[unit_index, ...] = self.sparsity.densify_waveforms(waveforms=waveforms, unit_id=unit_id)
+
+        return dense_waveforms
+
+    def get_sparse_templates(self) -> np.ndarray:
+        # Objects without sparsity mask don't have sparsity and therefore can't return sparse templates
+        if self.sparsity is None:
+            raise ValueError("Can't return sparse templates without passing a sparsity mask")
+
+        # Waveforms are already sparse
+        if not self.sparsity.are_waveforms_dense(self.templates_array):
+            return self.templates_array
+
+        max_num_active_channels = self.sparsity.max_num_active_channels
+        sparse_waveforms = np.zeros(shape=(self.num_units, self.num_samples, max_num_active_channels))
+        for unit_index, unit_id in enumerate(self.unit_ids):
+            waveforms = self.templates_array[unit_index, ...]
+            sparse_waveforms[unit_index, ...] = self.sparsity.sparsify_waveforms(waveforms=waveforms, unit_id=unit_id)
+
+        return sparse_waveforms
+
+    def are_templates_sparse(self) -> bool:
+        if self.sparsity is None:
+            return False
```

Comment:
```suggestion
        return self.sparsity is not None
```
Otherwise it would return `None`, right?